domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmChildAddHandler.java

```diff
@@ -56,6 +57,11 @@ public SecurityRealmChildAddHandler( boolean validateAuthentication, boolean val
     }
 
     @Override
+    protected boolean isResourceServiceRestartAllowed(OperationContext context, ServiceController<?> service) {
+        return false;
+    }
```

Comment:
@darranl confirmed what I found experimenting: there are some services that are not declaring a dependency on the Security realm and are expecting it to be up.